cpp/benchmarks/search/contains.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+
+#include <cudf/detail/search.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <cudf/types.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+namespace {
+template <typename Type>
+std::unique_ptr<cudf::table> create_table_data(cudf::size_type n_rows,
+                                               cudf::size_type n_cols,
+                                               bool has_nulls = false)
+{
+  data_profile profile;
+  profile.set_cardinality(0);
+  profile.set_null_frequency(has_nulls ? 0.1 : 0.0);
```

Comment:
Be aware that `0.0` will still set the null mask with all bits in mask as valid. If you prefer not to have null mask, you should pass `std::nullopt`.
```
  profile.set_null_frequency(has_nulls ? std::optional{0.1} : std::nullopt);
```